### ISSUE TITLE
[BUG] `SpecialUnitary` is not trainable with autodiff on `"default.mixed"` device

### ISSUE THREAD

**Issue by dwierichs** [PennyLaneAI/pennylane#3793]
### Expected behavior

The following code should produce a non-vanishing gradient:

<details>
<summary>Code</summary>

```python
dev = qml.device('default.mixed', wires=1)

@qml.qnode(dev, interface="jax")
def circuit(x):
    qml.SpecialUnitary(x, wires=[0])
    return qml.expval(qml.PauliZ(0))

x = jax.numpy.array([0.2, 0.1, 0.6])
jax.grad(circuit)(x) # Should produce Array([-0.68453361, -0.3422668 ,  0.03581418], dtype=float64)
```

</details>

### Actual behavior

The above code exampe produces a vanishing gradient:

<details>
<summary>Code</summary>

```python
dev = qml.device('default.mixed', wires=1)

@qml.qnode(dev, interface="jax")
def circuit(x):
    qml.SpecialUnitary(x, wires=[0])
    return qml.expval(qml.PauliZ(0))

x = jax.numpy.array([0.2, 0.1, 0.6])
```
```pycon
>>> jax.grad(circuit)(x)
Array([0., 0., 0.], dtype=float64)
```

</details>

### Additional information

This is caused by the following pipeline:
- `"default.mixed"` device does not support `SpecialUnitary`, because it is not registered with the device, and other than `"default.qubit"`, having a matrix is not sufficient for an operation to be supported
- `SpecialUnitary` is being decomposed, transferring the trainability of its parameters to `TmpPauliRot` operations, which have parameters 0.
- `TmpPauliRot` again is not supported (which is intended) and also being decomposed. As the `TmpPauliRot` operations have zero parameter, they are removed in this decomposition step, preventing the gradient computation from succeeding.

For `"default.mixed"`, and in fact any device that can deal with operations that provide a matrix, adding `SpecialUnitary` to the supported operations is sufficient.

For other devices, we might run into trouble in the future if a custom device:
- is to be used with automatic differentiation (i.e. not with a gradient transform)
- can not handle operations that provide a matrix generically

### System information

```shell
Name: PennyLane
Version: 0.29.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/david/venvs/dev/lib/python3.10/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning, PennyLane-qiskit

Platform info:           Linux-5.15.0-60-generic-x86_64-with-glibc2.35
Python version:          3.10.6
Numpy version:           1.23.5
Scipy version:           1.9.3
Installed devices:
- default.gaussian (PennyLane-0.29.0.dev0)
- default.mixed (PennyLane-0.29.0.dev0)
- default.qubit (PennyLane-0.29.0.dev0)
- default.qubit.autograd (PennyLane-0.29.0.dev0)
- default.qubit.jax (PennyLane-0.29.0.dev0)
- default.qubit.tf (PennyLane-0.29.0.dev0)
- default.qubit.torch (PennyLane-0.29.0.dev0)
- default.qutrit (PennyLane-0.29.0.dev0)
- null.qubit (PennyLane-0.29.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.28.2)
- qiskit.aer (PennyLane-qiskit-0.28.0)
- qiskit.basicaer (PennyLane-qiskit-0.28.0)
- qiskit.ibmq (PennyLane-qiskit-0.28.0)
- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.28.0)
- qiskit.ibmq.sampler (PennyLane-qiskit-0.28.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by Qottmann** [comment]
Is there a particular reason that `default.mixed` does not support operations that provide a matrix like `default.qubit` does?

**Comment by dwierichs** [comment]
I don't know, to be honest. @josh146, do you?

**Comment by josh146** [comment]
`default.qubit` supporting operations that have `matrix` defined is a new feature we added recently (several releases ago) 🙂 

We were actually very cautious and only added it after a lot of benchmarking --- there was a fear that it could cause performance regressions; there are operations (such as QFT) where it is more efficient to decompose than use the matrix.

**Comment by josh146** [comment]
[In fact, iirc it _did_ cause performance regressions, so it is a tradeoff --- more functionality at a cost of reduction in performance in some workflows]

**Comment by dwierichs** [comment]
Sounds like we want to stick to manual additions to the class attribute `operations` for now, right? :)
